### ISSUE TITLE
"Hide private files" button for Search Files

### ISSUE THREAD

**Issue by redactedscribe** [nicotine-plus/nicotine-plus#1737]
### Describe the solution you'd like
A button next to Free Slot could be introduced: "Hide private files". Being a filter, it could toggle private file visibility. Why would this be useful? "Show privately shared files in search results" could be enabled, as well as both "Enable search result filters by default" and "Hide private files". Then, any time you wish to inspect your results for private file results, we could just clear the filter / untoggle the button.

My thinking is most of the time I am not concerned with private files, but it's more convenient to already have the results returned than to have to go to the preferences and enable "Show privately shared files in search results" and then reperform the search.

**Comment by slook** [comment]
It sounds useful, but the multiple layers of filtering happening here could lead to confusion.

**Comment by redactedscribe** [comment]
That's true; it could end up being too complex if it's not obvious enough how it all works. Main point to understand would be with the Preference option disabled, the results you receive won't ever contain private files, so there is nothing to filter or toggle to reveal them.

**Comment by slook** [comment]
Then if the Preference option is disabled, the proposed toggle button should be hidden. Since the filter bar is already rather crowded, it would be good idea to be able to hide/show the filter widgets if certain Result Filters are not required/desired.

If options for the visibility of Filter widgets are added into Preferences>Searches and/or a new context menu for the Result Filters bar, then it would be a good idea for the Private visibility toggle item to be hidden/disabled/greyed-out according to if the global preference is active or not, this would help to avoid confusion between the conflicting options.

**Comment by mathiascode** [comment]
Just a general note: Please don't start implementing anything before a consensus is reached, otherwise work will probably be wasted. Anything filter/column-related quickly becomes convoluted, and I want to avoid unnecessary clutter.

**Comment by slook** [comment]
I'm not going to be implementing anything related to this, since it's beyond my capability and involves core changes with TreeView etc, which I know you already have long-term plans for to make into a custom class, it's best if you do it the way you want/need to, because that will be a lot cleaner than anything I could come up with.

**Comment by slook** [comment]
This request is blocked until we've ported the search result list to the new TreeView class.

The suggested filter will depend on #1736

**Comment by mathiascode** [comment]
@redactedscribe Does https://github.com/nicotine-plus/nicotine-plus/pull/3379 work as expected?

Downloads at the bottom of this page: https://github.com/nicotine-plus/nicotine-plus/actions/runs/14955405478

**Comment by mathiascode** [comment]
Thanks for testing!

> Took a couple years, but thanks for implementing this [@mathiascode](https://github.com/mathiascode).

Yeah, a lot of time was spent on the development of and bug fix releases for 3.3.x (about three years in total). I'm hoping we can get 3.4.0 out sooner with a bunch of smaller improvements like this.
